### PR TITLE
Add internal api ApiWbStackSiteStatsUpdate

### PR DESCRIPTION
**Pull request by toban**
This runs the initSiteStats.php for requested wiki

### CODE REVIEW

File: dist-persist/wbstack/src/Internal/ApiWbStackSiteStatsUpdate.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class ApiWbStackSiteStatsUpdate extends \ApiBase {
+    public function mustBePosted() {return true;}
+    public function isWriteMode() {return true;}
+    public function isInternal() {return true;}
+    public function execute() {
+        global $IP;
+
+        @set_time_limit( 60*5 ); // 5 mins maybe D:
+		@ini_set( 'memory_limit', '-1' ); // also try to disable the memory limit? Is this even a good idea?
```
Comment (deer-wmde) [inline dist-persist/wbstack/src/Internal/ApiWbStackSiteStatsUpdate.php:16]: I'm not really a fan of using these functions here, and also suppressing errors - I guess it's there so the output doesn't get garbled and the request doesn't time out, so it makes sense but I think it is a bit of bad practice.

That said, I don't have an alternative approach at hand right now so there's that :sweat_smile:

Comment (toban) [reply dist-persist/wbstack/src/Internal/ApiWbStackSiteStatsUpdate.php:16]: So this is used(copy-pasta) for most of these endpoints that do any heavy lifting. If you want we could try removing the memory limit for example, not sure if it's needed. The time limit i think is required though.

What would be suppressing errors?

Comment (deer-wmde) [reply dist-persist/wbstack/src/Internal/ApiWbStackSiteStatsUpdate.php:16]: > What would be suppressing errors?

The `@` operator in front of function calls

> So this is used(copy-pasta) for most of these endpoints that do any heavy lifting. 

Ah I wasn't aware, then it's more of a refactoring thing I guess, I'm okay with that :+1:

Comment (toban) [reply dist-persist/wbstack/src/Internal/ApiWbStackSiteStatsUpdate.php:16]: Huh, I did not know that was for error control. TIL!

File: dist-persist/wbstack/src/loadInternal.php [hunk 2]
```diff
@@ -24,6 +25,7 @@
 $wgAPIModules['wbstackElasticSearchInit'] = \WBStack\Internal\ApiWbStackElasticSearchInit::class;
 $wgAPIModules['wbstackForceSearchIndex'] = \WBStack\Internal\ApiWbStackForceSearchIndex::class;
 $wgAPIModules['wbstackQueueSearchIndexBatches'] = \WBStack\Internal\ApiWbStackQueueSearchIndexBatches::class;
+$wgAPIModules['wbstacksitestatsupdate'] = \WBStack\Internal\ApiWbStackSiteStatsUpdate::class;
```
Comment (deer-wmde) [inline dist-persist/wbstack/src/loadInternal.php:28]:
```suggestion
$wgAPIModules['wbstackSiteStatsUpdate'] = \WBStack\Internal\ApiWbStackSiteStatsUpdate::class;
```
Nothing dramatic but I see the other module names are written camelCase, could we use it here too?

Comment (toban) [reply dist-persist/wbstack/src/loadInternal.php:28]: Yeah sure will probably need to change that on the api too

Comment (toban) [reply dist-persist/wbstack/src/loadInternal.php:28]: these https://github.com/wbstack/api/pull/499/files#diff-5a8ae3065e9b4af4cbba73b1b3b35752a60680bd5d0372f8ebee122ab8581857R70

**Pull request merged**